src/Bicep.LangServer/Configuration/JsonEditor.cs

```diff
@@ -0,0 +1,221 @@
+                }
+                else
+                {
+                    return null;
+                }
+            }
+
+            return null;
+        }
+
+        public static string ApplyInsertion(string text, (int line, int column, string text) insertion)
+        {
+            var lineStarts = TextCoordinateConverter.GetLineStarts(text);
+            int offset = TextCoordinateConverter.GetOffset(lineStarts, insertion.line, insertion.column);
+            return text.Substring(0, offset) + insertion.text + text.Substring(offset);
+        }
+
+        private (int line, int column, string insertText)? AppendToEndOfJson(string text)
+        {
+            var lastLine = _lineStarts.Length - 1;
+            var lastLineLength = _json.Length - _lineStarts[_lineStarts.Length - 1];
+            Debug.Assert(lastLine >= 0 && lastLine < _lineStarts.Length, $"{nameof(lastLine)} out of bounds");
+            Debug.Assert(lastLineLength >= 0, $"{nameof(lastLineLength)} out of bounds");
+
+            if (lastLineLength > 0)
+            {
+                text = "\n" + text;
+            }
+
+            return (lastLine, lastLineLength, text);
+        }
+
+        private string PopFromLeft(List<string> list)
+        {
+            if (list.Count == 0)
+            {
+                throw new ArgumentException($"{nameof(list)} should not be empty");
+            }
+
+            string value = list[0];
+            list.RemoveAt(0);
+            return value;
+        }
+
+        private object PropertyPathToObject(string[] propertyPaths, object propertyValue)
+        {
+            List<string> remainingPaths = new(propertyPaths);
+            remainingPaths.Reverse();
+            Dictionary<string, object> result = new();
+            string leafName = PopFromLeft(remainingPaths);
+            result[leafName] = propertyValue;
+
+            foreach (string propertyName in remainingPaths)
+            {
+                Dictionary<string, object> embeddedObject = new();
+                embeddedObject[propertyName] = result;
+                result = embeddedObject;
+            }
+
+            return result;
+        }
+
+        private string Stringify(object value, int indent)
+        {
+            StringBuilder sb = new StringBuilder();
+            TextWriter textWriter = new StringWriter(sb);
+            JsonTextWriter jsonWriter = new(textWriter);
+            jsonWriter.Formatting = Formatting.Indented;
+            jsonWriter.IndentChar = ' ';
+            jsonWriter.Indentation = _indent;
+
+            new JsonSerializer().Serialize(jsonWriter, value);
+
+            return sb.ToString();
+        }
+
+        private string IndentEachLine(string text, int indent)
+        {
+            string indentString = new string(' ', indent);
+            return indentString + text.Replace("\n", "\n" + indentString);
+        }
+
+        private int GetIndentationOfLine(int line)
```

Comment:
Suggested change:
```
        private int GetIndentationOfLine(int line)
        {
            int currentLineStart = _lineStarts[line];

            if (_lineStarts.Length > line + 1)
            {
                int nextLineStart = _lineStarts[line + 1];

                var textBetweenCurrentAndNextLineStart = _json.Substring(currentLineStart, nextLineStart - currentLineStart);

                return textBetweenCurrentAndNextLineStart.Length - textBetweenCurrentAndNextLineStart.TrimStart(' ').Length;
            }

            return 0;
        }
```

Reply:
Good idea, but this one is tested and working, should be sufficient.